packages/arb-bridge-eth/contracts/vm/NodeGraph.sol

```diff
@@ -107,23 +107,45 @@ contract NodeGraph is ChallengeType {
         bytes32 logsAccHash;
     }
 
-    function pruneLeaf(
-        bytes32 from,
-        bytes32[] calldata leafProof,
-        bytes32[] calldata latestConfirmedProof
+    function pruneLeaves(
+        bytes32[] memory fromNodes,
+        bytes32[] memory leafProofs,
+        uint256[] memory leafProofLengths,
+        bytes32[] memory latestConfProofs,
+        uint256[] memory latestConfirmedProofLengths
     )
-        external
+        public
     {
-        bytes32 leaf = RollupUtils.calculatePath(from, leafProof);
-        require(isValidLeaf(leaf), PRUNE_LEAF);
+        uint pruneCount = fromNodes.length;
+
         require(
-            leafProof[0] != latestConfirmedProof[0] &&
-            RollupUtils.calculatePath(from, latestConfirmedProof) == latestConfirmed(),
-            PRUNE_CONFLICT
+            leafProofLengths.length == pruneCount &&
+            latestConfirmedProofLengths.length == pruneCount,
+            "input length mistmatch"
         );
-        delete leaves[leaf];
-
-        emit RollupPruned(leaf);
+        uint256 prevLeafOffset = 0;
+        uint256 prevConfOffset = 0;
+        for (uint256 i = 0; i < pruneCount; i++) {
+            bytes32 from = fromNodes[i];
```

Comment:
I think we need to add `require(leafProofLengths[i] > 0 && latestConfirmedProofLengths[i] > 0)` here.

Reply:
Fixed this. Will push soon.